pkg/ddc/alluxio/transform_ufs.go

```diff
@@ -45,10 +45,21 @@ func (e *AlluxioEngine) transformDatasetToVolume(runtime *datav1alpha1.AlluxioRu
 				value.UFSVolumes = []UFSVolume{}
 			}
 
-			value.UFSVolumes = append(value.UFSVolumes, UFSVolume{
-				Name:          strings.TrimPrefix(mount.MountPoint, common.VolumeScheme.String()),
-				ContainerPath: utils.UFSPathBuilder{}.GenLocalStoragePath(mount),
-			})
+			// if mount.MountPoint contains subpath
+			if strings.Count(mount.MountPoint, "/") > 2 {
+				pvc := strings.Split(strings.TrimPrefix(mount.MountPoint, common.VolumeScheme.String()), "/")
+
+				value.UFSVolumes = append(value.UFSVolumes, UFSVolume{
+					Name:          pvc[0],
+					SubPath:       strings.Join(pvc[1:], "/"),
+					ContainerPath: utils.UFSPathBuilder{}.GenLocalStoragePath(mount),
+				})
+			} else {
+				value.UFSVolumes = append(value.UFSVolumes, UFSVolume{
+					Name:          strings.TrimPrefix(mount.MountPoint, common.VolumeScheme.String()),
+					ContainerPath: utils.UFSPathBuilder{}.GenLocalStoragePath(mount),
+				})
+			}
```

Comment:
I suggest using the `strings.SplitN` function to split the MountPoint into two parts. This can handle cases where MountPoint contains multiple '/' characters and avoid unnecessary string operations. Additionally, it is recommended to add necessary comments to each UFSVolume to improve the code's readability.

```golang
// Split MountPoint into PVC name and subpath (if it contains a subpath)
parts := strings.SplitN(strings.TrimPrefix(mount.MountPoint, common.VolumeScheme.String()), "/", 2)

if len(parts) > 1 {
    // MountPoint contains subpath
    value.UFSVolumes = append(value.UFSVolumes, UFSVolume{
        Name:          parts[0],
        SubPath:       parts[1],
        ContainerPath: utils.UFSPathBuilder{}.GenLocalStoragePath(mount),
        // Add additional comments as needed
    })
} else {
    // MountPoint does not contain subpath
    value.UFSVolumes = append(value.UFSVolumes, UFSVolume{
        Name:          parts[0],
        ContainerPath: utils.UFSPathBuilder{}.GenLocalStoragePath(mount),
        // Add additional comments as needed
    })
}

```

Reply:
ok!